homeassistant/components/tradfri/light.py

```diff
@@ -49,12 +49,11 @@ async def async_setup_entry(
     api = tradfri_data[KEY_API]
     devices = tradfri_data[DEVICES]
 
-    lights = [dev for dev in devices if dev.has_light_control]
-    if lights:
-        async_add_entities(TradfriLight(light, api, gateway_id) for light in lights)
-
+    async_add_entities(
+        TradfriLight(dev, api, gateway_id) for dev in devices if dev.has_light_control
+    )
     if config_entry.data[CONF_IMPORT_GROUPS] and (groups := tradfri_data[GROUPS]):
-        async_add_entities(TradfriGroup(group, api, gateway_id) for group in groups)
+        async_add_entities([TradfriGroup(group, api, gateway_id) for group in groups])
```

Comment:
It would be nice to extend the light list with the group list and only do one call to `async_add_entities` at the end.

Reply:
Done.

Line 56 need the [], otherwise .extend fails when there are only 1 group